go/arrow/float16/float16.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package float16
+
+import (
+	"math"
+)
+
+type Num uint16
+
+// https://en.wikipedia.org/wiki/Half-precision_floating-point_format
+func NewFloat16(f float32) Num {
```

Comment:
I would perhaps rename this as just `New`.
in user code it will look like:
`float16.New(42)` instead of the unnecessary stutter `float16.NewFloat16(42)`.